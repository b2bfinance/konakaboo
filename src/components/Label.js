import {
  blue,
  blueGrey,
  orange,
  red,
  teal,
  yellow
} from '@material-ui/core/colors';
import { getContrastRatio } from '@material-ui/core/styles/colorManipulator';
import Typography from '@material-ui/core/Typography';
import React from 'react';
import styled from 'styled-components';

export const colors = {
  blue,
  blueGrey,
  red,
  orange,
  yellow,
  teal
};

export const Label = styled.div`
  border-radius: 1rem;
  padding: 0.2rem 0.75rem;
  background-color: ${props => props.colorGroup[600]};
  color: ${props =>
    getContrastRatio(props.colorGroup[600], props.colorGroup[50]) >= 3
      ? props.colorGroup[50]
      : colors.blueGrey[800]};
`;

export const getLabel = label => {
  if (label.startsWith('#')) {
    const firstSpaceIndex = label.indexOf(' ');
    return [label.slice(1, firstSpaceIndex), label.slice(firstSpaceIndex + 1)];
  }

  return ['blueGrey', label];
};

export default ({ label }) => {
  const [color, text] = getLabel(label);

  return (
    <Label colorGroup={colors[color] || colors.blueGrey}>
      <Typography color="inherit" variant="caption">
        {text}
      </Typography>
    </Label>
  );
};
