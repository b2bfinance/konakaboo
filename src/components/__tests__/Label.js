import React from 'react';
import { render } from 'enzyme';
import Label from '../Label';

test('<Label /> renders correctly', () => {
  expect(render(<Label />)).toMatchSnapshot();
});
