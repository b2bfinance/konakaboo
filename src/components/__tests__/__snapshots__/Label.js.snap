// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> renders correctly with an unsupported color label 1`] = `
.c0 {
  border-radius: 1rem;
  padding: 0.2rem 0.75rem;
  background-color: #546e7a;
  margin-right: 0.25rem;
  color: #eceff1;
}

.c0:last-child {
  margin-right: 0;
}

<div
  class="c0"
>
  <span
    class="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-colorInherit-30"
  >
    Label
  </span>
</div>
`;

exports[`<Label /> renders correctly with blue color label 1`] = `
.c0 {
  border-radius: 1rem;
  padding: 0.2rem 0.75rem;
  background-color: #1e88e5;
  margin-right: 0.25rem;
  color: #e3f2fd;
}

.c0:last-child {
  margin-right: 0;
}

<div
  class="c0"
>
  <span
    class="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-colorInherit-30"
  >
    Label
  </span>
</div>
`;

exports[`<Label /> renders correctly with blueGrey color label 1`] = `
.c0 {
  border-radius: 1rem;
  padding: 0.2rem 0.75rem;
  background-color: #546e7a;
  margin-right: 0.25rem;
  color: #eceff1;
}

.c0:last-child {
  margin-right: 0;
}

<div
  class="c0"
>
  <span
    class="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-colorInherit-30"
  >
    Label
  </span>
</div>
`;

exports[`<Label /> renders correctly with colored label 1`] = `
.c0 {
  border-radius: 1rem;
  padding: 0.2rem 0.75rem;
  background-color: #e53935;
  margin-right: 0.25rem;
  color: #ffebee;
}

.c0:last-child {
  margin-right: 0;
}

<div
  class="c0"
>
  <span
    class="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-colorInherit-30"
  >
    Label
  </span>
</div>
`;

exports[`<Label /> renders correctly with default label 1`] = `
.c0 {
  border-radius: 1rem;
  padding: 0.2rem 0.75rem;
  background-color: #546e7a;
  margin-right: 0.25rem;
  color: #eceff1;
}

.c0:last-child {
  margin-right: 0;
}

<div
  class="c0"
>
  <span
    class="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-colorInherit-30"
  >
    Label
  </span>
</div>
`;

exports[`<Label /> renders correctly with orange color label 1`] = `
.c0 {
  border-radius: 1rem;
  padding: 0.2rem 0.75rem;
  background-color: #fb8c00;
  margin-right: 0.25rem;
  color: #37474f;
}

.c0:last-child {
  margin-right: 0;
}

<div
  class="c0"
>
  <span
    class="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-colorInherit-30"
  >
    Label
  </span>
</div>
`;

exports[`<Label /> renders correctly with red color label 1`] = `
.c0 {
  border-radius: 1rem;
  padding: 0.2rem 0.75rem;
  background-color: #e53935;
  margin-right: 0.25rem;
  color: #ffebee;
}

.c0:last-child {
  margin-right: 0;
}

<div
  class="c0"
>
  <span
    class="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-colorInherit-30"
  >
    Label
  </span>
</div>
`;

exports[`<Label /> renders correctly with teal color label 1`] = `
.c0 {
  border-radius: 1rem;
  padding: 0.2rem 0.75rem;
  background-color: #00897b;
  margin-right: 0.25rem;
  color: #e0f2f1;
}

.c0:last-child {
  margin-right: 0;
}

<div
  class="c0"
>
  <span
    class="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-colorInherit-30"
  >
    Label
  </span>
</div>
`;

exports[`<Label /> renders correctly with yellow color label 1`] = `
.c0 {
  border-radius: 1rem;
  padding: 0.2rem 0.75rem;
  background-color: #fdd835;
  margin-right: 0.25rem;
  color: #37474f;
}

.c0:last-child {
  margin-right: 0;
}

<div
  class="c0"
>
  <span
    class="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-colorInherit-30"
  >
    Label
  </span>
</div>
`;
